Support zend-servicemanager v2+v3

weierophinney opened the pull request:
This patch builds on the one originally submitted via #21, and expands it to provide support for both v2 and v3 of zend-servicemanager. Specifically, it introduces the following changes:
- `composer.json`:
  - Updates PHP version to `^5.5 || ^7.0`
  - Updates develop branch alias to `2.6`
  - Updates zend-servicemanager requirement to `^2.7.2 || ^3.0`
- Code changes:
  - Updated `LoaderPluginManager` to implement both `validate()` and `validatePlugin()` (latter proxies to former).
  - Updated `TranslatorServiceFactory` to implement both v2 and v3 factory methods, with v2 method proxying to v3.
  - Updated `HelperConfig`:
    - define aliases + invokable factories;
  - alter `configureServiceManager` to call SM methods setters instead of `configure()` (for portability)
  - implemented `toArray()` (v3 method)
  - Updated `TranslatorTest` to use Config objects + `configureServiceManager()` method to seed SM and plugin manager instances.

ezimuel committed:
Fixed the plugin with the new Service Manager version

weierophinney committed:
Merge pull request #21 from ezimuel/fix/plugin-service-manager

Fixed the plugin with the new Service Manager version

weierophinney committed:
Complete updates to support zend-servicemanager v2 + v3

- `composer.json`:
  - Updates PHP version to `^5.5 || ^7.0`
  - Updates develop branch alias to `2.6`
  - Updates zend-servicemanager requirement to `^2.7.2 || ^3.0`
- Code changes:
  - Updated `LoaderPluginManager` to implement both `validate()` and
    `validatePlugin()` (latter proxies to former).
  - Updated `TranslatorServiceFactory` to implement both v2 and v3 factory
    methods, with v2 method proxying to v3.
  - Updated `HelperConfig`:
    - define aliases + invokable factories;
    - alter `configureServiceManager` to call SM methods setters instead of
      `configure()` (for portability)
    - implemented `toArray()` (v3 method)
  - Updated `TranslatorTest` to use Config objects + `configureServiceManager()`
    method to seed SM and plugin manager instances.

weierophinney committed:
Added 2.7/3.0-specific builds to the test matrix

weierophinney commented:
Labeled as a WIP, as several dependencies are on components that also bring in SM refactors, but which have not been updated to support both v2 and v3 of the SM (only v3). As such, the CI checks on this will fail until those are updated.

ezimuel commented:
@weierophinney can you update the composer.json and .travis to use zend-servicemanager 2.7.3?

weierophinney committed:
Updated zend-eventmanager dependency

Use v2.6.2 || v3.0.

weierophinney committed:
Ensure event manager and service manager are tested on both v2 and v3.

Adds the pattern used in zendframework/zend-serializer#8 for determining
which version of each library to use when testing.

weierophinney commented:
@ezimuel Done! I made it test against the v2 and v3 versions of _both_ the service manager and event manager while I was at it.

weierophinney committed:
Updated minimum servicemanager version in composer.json

Travis config is already correct.

weierophinney committed:
Pin to stable versions

Pins all requirements to stable versions, selecting versions
forwards-compatible with zend-stdlib and zend-servicemanager 3.0 where
available. For those that are not, we will omit them from testing when testing
forwards compatibility.

weierophinney committed:
Conditionally omit tests based on forwards compatibility status

Omits tests utilizing features from zend-view, zend-validator, and zend-cache
when their classes are unavailable, and omits installing those components when
performing forwards-compatibility tests for zend-servicemanager v3.

weierophinney committed:
Ensure forwards compatibility of aliases

Adds `LoaderPluginManagerCompatibilityTest` to allow ensuring the
`LoaderPluginManager` is forwards compatible with zend-servicemanager v3, and
fixes several inconsistencies.

Additionally, updates the `Zend\I18n\View\HelperConfig::$factories` definition
to add the legacy (v2) canonicalized helper keys required for plugin managers to
remain v2 compatible.

weierophinney committed:
Fixed errors when running against v2 components

- Fixed argument name for `LoaderPluginManager::validatePlugin()`
  implementation so that it's consistent with the definition.
- Rewrote `triggerUntil()` methods as `triggerEventUntil()`, as the
  `triggerUntil()` argument order changes between v2 and v3.
- Use the `config` service, not the `Config` or `Configuration` service.
- Mock against container-interop, not `ServiceLocatorInterface`

weierophinney committed:
CS fixes per php-cs-fixer
